controllers/storageclient/interface.go

```diff
@@ -1,10 +1,18 @@
 // Package storageclient provides the storage client for the controllers package
 package storageclient
 
+import "github.com/iter8-tools/iter8/base/summarymetrics"
+
+// versionMetricSummary is a map of metric names to summary metric values
+type versionMetricSummary map[string]summarymetrics.SummaryMetric
+
 // Interface enables interaction with a storage entity
 // Can be mocked in unit tests with fake implementation
 type Interface interface {
-	// CreateMetric(name string, value interface{}) error
-	// GetMetric(name string) (interface{}, error)
-	// DeleteMetric(name string) (interface{}, error)
+	// WriteMetric is called by the A/B/n SDK gRPC API implementation (SDK for application clients)
+	WriteMetric(applicationName string, version string, user string, metricName string, metricValue float64) error
+
```

Comment:
GetSignature(applicationName string, version int) and SetSignature(applicationName string, version int, signature string) for the purpose of getting prefixes for iteration

Key 1: signature::my-app::0 (get the signature of the last version)
Key 2: metric::my-app::0::my-signature::my-metric::my-user (get the metric value with all the provided information)
Key 3: app-metrics::my-app (get a list of metrics associated with my-app)
Key 4: app-versions::my-app (get a number of versions for my-app)

Reply:
`SetKey5 call is useful` -- will be invoked by lookup.

Reply:
Support Set for all the above Keys. And support `GetSummaryMetrics`. The latter might need other calls, based on its implementation design (which we're yet to do).